auto_prove_sequent.ml

```diff
@@ -1,13 +1,24 @@
+let try_auto_prover_on_sequent sequent =
+    try Some (Focused_proof.prove_sequent sequent), true
+    with Focused_proof.NonProvableSequent -> None, false
+         | Focused_proof.NonAutoProvableSequent -> None, true
+
 let auto_prove_sequent_with_exceptions request_as_json =
-    let sequent = Raw_sequent.sequent_from_json request_as_json in
-    try Proof.auto_weak sequent
+    let sequent_with_notations = Sequent_with_notations.from_json request_as_json in
+    try Some (Proof.auto_weak sequent_with_notations.sequent), true
     with Proof.AutoWeakNotApplicable ->
-        Focused_proof.prove_sequent sequent;;
+        let result = try_auto_prover_on_sequent sequent_with_notations.sequent in
+        match result with
+        | Some proof, _ -> result
+        | None, _ ->
+            let cyclic_notations, acyclic_notations = Sequent_with_notations.split_cyclic_acyclic sequent_with_notations in
+            if List.length cyclic_notations + List.length acyclic_notations > 0
+            then None, true
```

Comment:
explain in documentation that orange ⊢ means either time out or cyclic notations

Reply:
Done. I will precise that it fails only with recursive notations, when the autoprover for non-recursive notations will be implemented.